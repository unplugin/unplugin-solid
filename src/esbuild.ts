/**
 * This entry file is for esbuild plugin. Requires esbuild >= 0.15
 *
 * @module
 */

import unplugin from '.'

/**
 * Esbuild plugin
 *
 * @example
 * ```ts
 * import { build } from 'esbuild'
 * import Solid from 'unplugin-solid/esbuild'
 *
 * build({ plugins: [Solid()] })
 ```
 */
const esbuild: typeof unplugin.esbuild = unplugin.esbuild
export default esbuild
export { esbuild as 'module.exports' }
