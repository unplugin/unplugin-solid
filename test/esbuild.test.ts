import { testFixtures } from '@sxzz/test-utils'
import { build } from 'esbuild'
import { describe, expect } from 'vitest'

import Solid from '../src/esbuild'

describe('esbuild', async () => {
  await testFixtures(
    'test/fixtures/*.{js,ts,jsx,tsx}',
    async (args, id) => {
      const result = await build({
        entryPoints: [id],
        bundle: true,
        external: ['solid-js'],
        treeShaking: true,
        format: 'esm',
        plugins: [
          Solid({
            dev: args.dev,
          }),
        ],
        write: false,
      })
      const codes = result.outputFiles.map((file) => file.text).join('\n')

      expect(
        codes
          .replaceAll(JSON.stringify(id), '"#FILE#"')
          .replaceAll('\0', '[NULL]'),
      ).toMatchSnapshot()
    },
    {
      params: [['dev', [true, false]]],
      promise: true,
    },
  )
})
