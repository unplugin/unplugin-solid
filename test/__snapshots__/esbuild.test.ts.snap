// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild > test/fixtures/basic.tsx > dev = false 1`] = `
"// test/fixtures/basic.tsx
import { template as _$template } from "solid-js/web";
var _tmpl$ = /* @__PURE__ */ _$template(\`<div>Hello World\`);
var Foo = _tmpl$();
var basic_default = Foo;
export {
  basic_default as default
};
"
`;

exports[`esbuild > test/fixtures/basic.tsx > dev = false 2`] = `undefined`;

exports[`esbuild > test/fixtures/basic.tsx > dev = true 1`] = `
"// test/fixtures/basic.tsx
import { template as _$template } from "solid-js/web";
var _tmpl$ = /* @__PURE__ */ _$template(\`<div>Hello World\`);
var Foo = _tmpl$();
var basic_default = Foo;
export {
  basic_default as default
};
"
`;

exports[`esbuild > test/fixtures/basic.tsx > dev = true 2`] = `undefined`;

exports[`esbuild > test/fixtures/syntax.tsx > dev = false 1`] = `
"// test/fixtures/syntax.tsx
import { template as _$template } from "solid-js/web";
import { delegateEvents as _$delegateEvents } from "solid-js/web";
import { createComponent as _$createComponent } from "solid-js/web";
import { createEffect, createSignal, onCleanup, onMount, Suspense } from "solid-js";
var _tmpl$ = /* @__PURE__ */ _$template(\`<button>+\`);
function Foo(props) {
  const [count, setCount] = createSignal(0);
  onMount(() => {
    console.log("mounted");
    onCleanup(() => {
      console.log("unmounted");
    });
  });
  createEffect(() => {
    console.log("count:", count());
  });
  return ["Children:", _$createComponent(Suspense, {
    fallback: "loading...",
    get children() {
      return props.children;
    }
  }), (() => {
    var _el$ = _tmpl$();
    _el$.$$click = () => setCount(count() + 1);
    return _el$;
  })()];
}
var syntax_default = Foo;
_$delegateEvents(["click"]);
export {
  syntax_default as default
};
"
`;

exports[`esbuild > test/fixtures/syntax.tsx > dev = false 2`] = `undefined`;

exports[`esbuild > test/fixtures/syntax.tsx > dev = true 1`] = `
"// test/fixtures/syntax.tsx
import { template as _$template } from "solid-js/web";
import { delegateEvents as _$delegateEvents } from "solid-js/web";
import { createComponent as _$createComponent } from "solid-js/web";
import { createEffect, createSignal, onCleanup, onMount, Suspense } from "solid-js";
var _tmpl$ = /* @__PURE__ */ _$template(\`<button>+\`);
function Foo(props) {
  const [count, setCount] = createSignal(0);
  onMount(() => {
    console.log("mounted");
    onCleanup(() => {
      console.log("unmounted");
    });
  });
  createEffect(() => {
    console.log("count:", count());
  });
  return ["Children:", _$createComponent(Suspense, {
    fallback: "loading...",
    get children() {
      return props.children;
    }
  }), (() => {
    var _el$ = _tmpl$();
    _el$.$$click = () => setCount(count() + 1);
    return _el$;
  })()];
}
var syntax_default = Foo;
_$delegateEvents(["click"]);
export {
  syntax_default as default
};
"
`;

exports[`esbuild > test/fixtures/syntax.tsx > dev = true 2`] = `undefined`;
